### ISSUE TITLE
Inquiry for PadChest test csv

### ISSUE THREAD

**Issue by cheliu-computation** [HieuPhan33/MAVL#2]
Dear Authors,

Thanks for your impressive work and valuable repo contribution!

After searching in the repo, I am able to reproduce results on CheXpert, RSNA,SIIM, ChestXray14, but unable to find the csv file for PadChest-seen, PadChest-Unseen, and PadChest-Rare.

It would be really grateful if you could provide the csv file for these three tasks.
Best Regards

**Comment by HieuPhan33** [comment]
Hi @cheliu-computation,

Thanks for being interested in our work. We have updated the data/padchest/test.csv.

Please note that we would like to update the AUC results of PadChest rare from 84.64% to 84.56%. We sincerely apologize for the typo. All other metrics on PadChest-seen, PadChest-rare and PadChest-unseen remain the same.

**Comment by cheliu-computation** [comment]
Thanks for your reply!

In the new uploaded csv, it only has two columns with 'image_path' and 'labels'.
May I ask how to distinguish which images are for PadChest-rare, and which for seen and unseen?

> Hi @cheliu-computation,
> 
> Thanks for being interested in our work. We have updated the data/padchest/test.csv.
> 
> Please note that we would like to update the AUC results of PadChest rare from 84.64% to 84.56%. Unfortunately, we lost the previous ckpt that produces the reported AUC score results on PadChest rare. We will keep searching and share with you and the public on such checkpoint when we found it.
> 
> All other metrics on PadChest-seen, PadChest-rare and PadChest-unseen are reproducible.

**Comment by HieuPhan33** [comment]
Hi,

The difference between PadChest-rare, PadChest-seen, and PadChest-unseen is from the classes we used for testing. They are all evaluated on the same 7961 images in the test.csv. The list of classes for PadChest-seen, PadChest-unseen, and PadChest-rare are provided from https://github.com/HieuPhan33/MAVL/blob/2f07deece4fc57a5147a540c8b772cd44907f3a7/Zero-shot_classification/test.py#L192 to https://github.com/HieuPhan33/MAVL/blob/2f07deece4fc57a5147a540c8b772cd44907f3a7/Zero-shot_classification/test.py#L210.


In other words, if the image has label 0 for all classes in PadChest-rare, it is still used to evaluate the model performance, i.e., AUC, ACC, and F1 scores to predict negative labels for PadChest-rare.
Indeed, this setting can lead to heavy class imbalance, especially for PadChest-rare and PadChest-unseen class evaluation.
We argue that this evaluation setting is more realistic, simulating the real use cases, where negative labels are more dominant among PadChest-rare disease classes.

**Comment by cheliu-computation** [comment]
Thanks for your prompt reply!
Amazing work, looking forward to your next work

**Comment by HieuPhan33** [comment]
As details about PadChest test.csv are provided, I will close this issue.

Please feel free to create another Github issue if you have any further requests.


Cheers!
